src/pages/articles/single.js

```diff
@@ -19,63 +18,53 @@ import request from 'constants/request.js';
-  const [repeatClaps, setRepeatClaps] = useState(0);
   const [clapCountAnimating, runClapCountAnimation] = useState(false);
+
   const dispatch = useDispatch();
-  const clapLimitPerArticle = useSelector(state => state.clapLimit[article.id]);
-  const limitReachedOnArticle = useSelector(state => state.limitReachedPerArticle);
- 
+
+  const numberOfClapsOnArticle = useSelector(
+    (state) => state.articleClapCount[article.id] || 0
+  );
+
   useEffect(() => {
     setLoaded(true);
   }, [isLoaded]);
 
   const incrementClapCount = () => {
-    
-    // Checks to see if user has accepted cookies before letting them like articles.
-    if(checkCookies('Accept cookies to clap')){
-      // Check if previous data has been stored on the article.
-      // If not allow user to clap, if yes check the claps have not exceeded the clap limit.
-      if (limitReachedOnArticle[article.id] == undefined || limitReachedOnArticle[article.id] <= clapLimit){
-    
-          // Increment claps and call action to change state in store.
-          setRepeatClaps(repeatClaps + 1);
-          // Increment counter for claps.
-          dispatch(clapStorageForArticles({
-              [article.id]: repeatClaps,
-          }));
-          // Storage for article that the user has maxed thier claps.
-          dispatch(clapLimitReachedStorage({[article.id]: repeatClaps}));
-
-      }else if( clapLimitPerArticle >= clapLimit || limitReachedOnArticle[article.id] >= clapLimit){
-       
-        // If clapLimit on the article has reached then the user cannot clap anymore.
-        // The second part of the else if block is useful only when the user revisits the article and tried to clap again.
-        dispatch(clapLimitReachedStorage({[article.id]: limitReachedOnArticle[article.id]}));
-        
-        alert.info(`That's enough clapping for today, buddy.`);
-        return;
-      }
-    
-      request({
-        url: `/api/v1/articles/${article.id}/clap`,
-        method: 'PUT',
-        headers: { Authorization: process.env.AUTH_KEY },
-        onSuccess: ({ claps }) => {
-          runClapCountAnimation(true);
-          setClapCount(claps);
-          setRepeatClaps(repeatClaps + 1);
-        }
-      });
+    // Prohibits clapping if user has not accepted cookie policy.
+    if (!checkCookies(prohibitedClapMessage)) return;
+
+    // Prohibit clapping if user has reached limit on claps.
+    if (numberOfClapsOnArticle >= clapLimit) {
+      alert.info(`You've reached your clap limit on this article, buddy.`);
+      return;
     }
-    
+
+    request({
+      url: `/api/v1/articles/${article.id}/clap`,
+      method: 'PUT',
+      headers: { Authorization: process.env.AUTH_KEY },
+      onSuccess: ({ claps }) => {
+        runClapCountAnimation(true);
+        setClapCount(claps);
+
+        dispatch(
+          incrementClapsOnArticle({
+            [article.id]: numberOfClapsOnArticle + 1
+          })
+        );
```

Comment:
I also moved the actual dispatch into the success of the request, because ideally, if the request to the server fails for one reason or another, it shouldn't actually proceed with incrementing the clap count.